### ISSUE TITLE
reshuffle of cellID from segmented tiff images with cajal.sample_seg

### ISSUE THREAD

**Issue by YuchenXiangEMBL** [CamaraLab/CAJAL#16]
Dear Patrick,

Thank you for the creating of CAJAL. It is a wonderful tool for neuron morphometry and I am excited to try it. I have segmented images and single-image-crops for my cells (both are tiff files), so I start from the cajal.sample_seg; however, I noticed that it relabelled the segmented images and I won't be able to find the corresponding cells for different clusters. 

Here follows my detailed code:
`import cajal.sample_seg`
`from os.path import join`
`bd='./Morphology//CAJAL/yfp_boost_test'`
`cajal.sample_seg.compute_icdm_all(infolder=join(bd,'corrected_segment_tiff'),out_csv=join(bd,'tiff_200pts_icdm.csv'),n_sample=200)` 
corrected_segment_tiff is the folder store all the segmented tiff images, where all the files should looks like '96-well-10X-2k-D7-001','96-well-10X-2k-D7-002'...
`import cajal.run_gw`
`cajal.run_gw.compute_gw_distance_matrix(join(bd,'tiff_200pts_icdm.csv'),join(bd,'tiff_200pts_gw_dmat.csv'),num_processes=8)`
`import cajal.utilities`
`from umap import umap_`
`import pandas as pd`
`cells, gw_dist_dict = cajal.utilities.read_gw_dists(join(bd,"tiff_200pts_gw_dmat.csv"), header=True)`
`gw_dist = cajal.utilities.dist_mat_of_dict(gw_dist_dict, cells)`
`reducer = umap_.UMAP(metric="precomputed", random_state=1)`
`embedding = reducer.fit_transform(gw_dist)`
`clusters = cajal.utilities.leiden_clustering(gw_dist, seed=1)`

`cells_cluster_9 = [n for m, n in zip(clusters, cells) if m==9]`
When I then call cells_cluter_9, I notice they return
`96-well-10X-2k-D7-001_0`

It is not directly readable with navis because it won't be able to find the path as it didn't exist
So I load my single image crop folder for reading, but I noticed that the cell label for my single image crop is not consistent with the cell label generated by CAJAL. 

I am wondering whether CAJAL does some re-labelling of individual cells the segmented tiff images instead of taking the cell label directly from the segmented tiff images. Because in this case I won't be able to visualize the cells in each clusters as I don't know there label as well. 

It would be wonderful if you can created a tutorial for showing us how to start from the segmented tiff images.

Thanks a lot,
Yuchen

**Comment by YuchenXiangEMBL** [comment]
I found a temporary solution to change the last line of cell_boundaries function to resolve the reshuffle problem. Although, for long-term use, it would be nice if it could be integrated this into code.
instead of 
`return list(enumerate(outlist))`
do 
`return [(cell_id, boundary_pts) for cell_id, boundary_pts in zip(cell_id_list, outlist)]`

**Comment by patrick-nicodemus** [comment]
Hi Yuchen, thanks for the comment. I'll take a look at this issue.

**Comment by patrick-nicodemus** [comment]
Yuchen, I agree with your comments and I will incorporate this fix or similar into CAJAL for the next release. Feel free to check back in if you don't see an update.
